File: 0167-two-sum-ii---input-array-is-sorted/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int,int> map;

        for(int i = 0; i < numbers.size(); i++){
            int comp = target - numbers[i];
            if(map.find(comp) != map.end()){
                return {map[comp], i+1};
            }
            else {
                map.insert(make_pair(numbers[i], i+1));
            }
            
        }
        return {};
    }
};

File: 0167-two-sum-ii---input-array-is-sorted/solution.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int c, index = 1;

        for(int i = 0; i<numbers.length; i++){
            c = target - numbers[i];
            if(map.containsKey(c)) return new int[]{map.get(c), index};
            else map.put(numbers[i], index);
            index++;
        }
        return null;
    }
}
